File: Sample/Sample/Settings View/AddNamedUserTableViewController.m
/* Copyright Airship and Contributors */

@import AirshipCore;
@import AirshipLocation;

#import "AddNamedUserTableViewController.h"

@implementation AddNamedUserTableViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.addNamedUserTextField.delegate = self;
}

- (void)setCellTheme {
    self.addNamedUserCell.backgroundColor = [UIColor whiteColor];
    self.addNamedUserTitle.textColor = [UIColor blackColor];
    self.addNamedUserTextField.textColor = [UIColor blackColor];
}

- (void)setTableViewTheme {
    self.tableView.backgroundColor = [UIColor whiteColor];
    
    [self.navigationController.navigationBar setTitleTextAttributes: @{NSForegroundColorAttributeName:[UIColor whiteColor]}];
    self.navigationController.navigationBar.barTintColor = [UIColor blackColor];
}

- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    [self setCellTheme];
    [self setTableViewTheme];
    
    if (UAirship.contact.namedUserID != nil) {
        self.addNamedUserTextField.text = UAirship.contact.namedUserID;
    }
}

- (BOOL)tableView:(UITableView *)tableView canEditRowAtIndexPath:(NSIndexPath *)indexPath {
    return NO;
}

- (BOOL)textFieldShouldReturn: (UITextField *)textField {
    if (textField.text != nil && textField.text.length > 0) {
        [UAirship.contact identify:textField.text];
    } else {
        [UAirship.contact reset];
    }
    
    [self.view endEditing:YES];
    
    [UAirship.channel updateRegistration];
    
    return [self.navigationController popViewControllerAnimated:YES] != nil;
}

@end

File: Airship/AirshipDebug/Source/DeviceInfo/AddNamedUserTableViewController.swift
/* Copyright Airship and Contributors */

import UIKit

#if canImport(AirshipCore)
import AirshipCore
#elseif canImport(AirshipKit)
import AirshipKit
#endif

class AddNamedUserTableViewController: UITableViewController, UITextFieldDelegate {

    @IBOutlet private weak var addNamedUserCell: UITableViewCell!
    @IBOutlet private weak var addNamedUserTitle: UILabel!
    @IBOutlet private weak var addNamedUserTextField: UITextField!

    override func viewDidLoad() {
        super.viewDidLoad()

        self.addNamedUserTextField.delegate = self
    }

    func setCellTheme() {
        addNamedUserCell.backgroundColor = ThemeManager.shared.currentTheme.Background
        addNamedUserTitle.textColor = ThemeManager.shared.currentTheme.PrimaryText
        addNamedUserTextField.textColor = ThemeManager.shared.currentTheme.PrimaryText
    }

    func setTableViewTheme() {
        tableView.backgroundColor = ThemeManager.shared.currentTheme.Background;
        navigationController?.navigationBar.titleTextAttributes = [.foregroundColor:ThemeManager.shared.currentTheme.NavigationBarText]
        navigationController?.navigationBar.barTintColor = ThemeManager.shared.currentTheme.NavigationBarBackground;

        addNamedUserTextField.attributedPlaceholder = NSAttributedString(string:"Named User", attributes: [NSAttributedString.Key.foregroundColor:ThemeManager.shared.currentTheme.SecondaryText])
    }

    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated);
        setCellTheme()
        setTableViewTheme()

        if let namedUserID = Airship.contact.namedUserID {
            addNamedUserTextField.text = namedUserID
        }
    }

    override func tableView(_ tableView: UITableView, canEditRowAt indexPath: IndexPath) -> Bool {
        return false
    }

    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
        if let namedUserID = textField.text, namedUserID.count > 0 {
            Airship.contact.identify(textField.text!)
        } else {
            Airship.contact.reset()
        }

        self.view.endEditing(true)
        
        Airship.push.updateRegistration()

        return navigationController?.popViewController(animated: true) != nil
    }
}
